File: Webservice/ws_sportFounder/SportFounderLibrary/Conversation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SportFounderLibrary
{
    public class Conversation
    {
        public int UserId { get; set; }
        public string UserNom { get; set; }
        public List<MessageSimple> Messages { get; set; }

        public Conversation() { }

        public Conversation(int userId)
        {
            UserId = userId;
            Messages = new List<MessageSimple>();
        }
    }
}

File: SportFounder2/app/src/main/java/com/example/jean_baptisteaniel/sportfounder2/Models/Conversation.java
package com.example.jean_baptisteaniel.sportfounder2.Models;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import org.json.JSONObject;

import java.util.List;

/**
 * Created by davy.azoulay on 29/07/2015.
 */
public class Conversation {
    private int UserId;
    private String UserNom;
    private List<MessageSimple> Messages;

    public Conversation(){}

    public static Conversation getConversationFromJson(JSONObject json){
        Gson gson = new GsonBuilder()
                .setDateFormat("yyyy'-'MM'-'dd'T'HH':'mm':'ss").create();
        return gson.fromJson(json.toString(), Conversation.class);
    }

    public int getUserId() {
        return UserId;
    }

    public void setUserId(int userId) {
        UserId = userId;
    }

    public String getUserNom() {
        return UserNom;
    }

    public void setUserNom(String userNom) {
        UserNom = userNom;
    }

    public List<MessageSimple> getMessages() {
        return Messages;
    }

    public void setMessages(List<MessageSimple> messages) {
        Messages = messages;
    }
}
